e2e/testdata/fn-eval/error-in-pipe/.expected/config.yaml

```diff
@@ -0,0 +1,20 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+exitCode: 1
+stdErr: |
+  error:   Stderr:
```

Comment:
What's the equivalent output when not piping?
Why/how this differs from that output?

Reply:
If there is only the first function in pipe, the output will be:
```
Stderr:
  "failed to configure function: input namespace cannot be empty"
Exit Code: 1
```
I guess we should output structured result to stderr when there is an error in the function in this case?

Reply:
Had an offline chat with @Shell32-Natsu about it. I think we can bring in consistency in the error output in `piped` and `non-piped` scenario. So if we have multiple `fn eval` chained, the error output need to have enough context such as "which function" and all the details such as structured-result output, stderr output dumped on `stderr` (something we already do in case of `non-piped`). He is going to share the proposed output. It will require minor tweak in our printing logic (`stderr` vs `stdout`).

Reply:
Expected output for kubeval:
```
[FAIL] "gcr.io/kpt-fn/kubeval:unstable"
  Results:
    [ERROR] selector is required in object "apps/v1/Deployment/" in file "resources.yaml" in field "selector"
    [ERROR] template is required in object "apps/v1/Deployment/" in file "resources.yaml" in field "template"
    [ERROR] Invalid type. Expected: [integer,null], given: string in object "apps/v1/Deployment/" in file "resources.yaml" in field "spec.replicas"
    [ERROR] Validating arbitrary CRDs is not supported yet. You can skip them by setting ignore_missing_schemas or skip_kinds in the function config
: ERR  - stdin: Failed initializing schema file:///tmp/master-standalone/custom-custom-v1.json: open /tmp/master-standalone/custom-custom-v1.json: n
o such file or directory  in object "custom.io/v1/Custom/" in file "resources.yaml"
  Exit code: 1
```

For set-namespace:
```
[FAIL] "gcr.io/kpt-fn/set-namespace:v0.1"
  Results:
    [ERROR] failed to configure function: input namespace cannot be empty
  Stderr:
    "failed to configure function: input namespace cannot be empty"
  Exit code: 1
```

WDYT? @frankfarzan @droot